### ISSUE TITLE
converting dtype in ImageCollection

### ISSUE THREAD

**Issue by tku137** [scikit-image/scikit-image#1438]
I am just getting into scikit image and came across some weird issue. I want to load a bunch of images as an ImageCollection, which works just fine. However, all images are loaded as uint8, but I need to have all images in the collection to be uint16.

When I try to use the example code from http://scikit-image.org/docs/dev/api/skimage.io.html#skimage.io.ImageCollection:
```python
def imread_convert(f):
    return imread(f).astype(np.uint8)

ic = ImageCollection('/tmp/*.png', load_func=imread_convert)
```
I get an error when trying to use the collection as normal, for example plot the data:
```
imread_convert() got an unexpected keyword argument 'img_num'
```
How is it possible to convert all images to another dtype? I tried simply mapping the utility-function and it works:
```python
ic = list(map(img_as_uint, ic))
```
Now I want to build a super class over ImageCollection and this solution is not applicable because (as far as I know) I would reassign `self`. I can't find the attribute where the data is stored, trying to
```python
self.data = list(map(img_as_uint, self.data))
```
yields to 
```
can not convert object to uint16
```
Sorry if this is a stupid question, I am shure I am missing something.

**Comment by jni** [comment]
Hi tku137!




I did a bit of digging for this one. It looks like your images are TIFFs, which have *support* for multi-frame files, even if it is unused. If your tiffs are single-page, then `img_num` will be passed to your `load_func` as just 0. Otherwise you will have (filename, 0), (filename, 1), ... and so on being read as frames in your image collection. See here:

https://github.com/scikit-image/scikit-image/blob/master/skimage/io/collection.py#L259





and here:


https://github.com/scikit-image/scikit-image/blob/master/skimage/io/collection.py#L202




The right thing to do will strongly depend on your files, but I hope this gets you on the right track. The easiest way is to ignore the img_num, but define your function as being able to take it:




```

def imread_convert(f, img_num=0):

    return imread(f).astype(np.uint16)

```




hope this helps!




Juan.




PS: Definitely not a stupid question! That was a subtle problem, and probably indicates we should update our docs!

**Comment by stefanv** [comment]
On Thu, Mar 19, 2015 at 4:19 PM, Juan Nunez-Iglesias <
notifications@github.com> wrote:

> PS: Definitely not a stupid question! That was a subtle problem, and
> probably indicates we should update our docs!


Please file an issue so that we don't forget!

**Comment by tku137** [comment]
Thank you very much for pointing this out. I'm using single png files, not tiff stacks. But setting `img_num=0` in the convert function works like a charm :)

Now I got everything the way I wanted it, thanks!

**Comment by blink1073** [comment]
I'm going to leave this open as a reminder to self to fix this.  What I propose is that `img_num==0`, we call just `load_func(f)`, since not all of the plugins accept `img_num` as a parameter.

**Comment by stefanv** [comment]
:+1:

**Comment by blink1073** [comment]
Fixed in #1440.
